.github/workflows/lint.yml

```
@@ -1,6 +1,10 @@
 name: Lint
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    branches: [main]
+  pull_request:
+  workflow_dispatch:
```

```suggestion
on: [push, pull_request, workflow_dispatch]
```

And also add `if: github.event_name != 'push' || github.event.repository.fork == true` under `lint:` below.